### PR TITLE
You can use a fishing rod to snatch organs during organ manipulation surgery (feat organs sanity fix)

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
This allows you to use a fishing rod during the "manipulate organs" step of the aforementioned surgery to snatch organs from a target.

Unlike other fish sources, this one has a negative fishing difficulty of -20, which when summed with the default minigame difficulty should still result in a negative difficulty. In layman terms, this means the minigame is skipped here (unless you're wearing some clunky stuff like insulated or boxing gloves). It also has a wait time of 8 to 13 seconds versus the more random standard 3 to 25 seconds.

A small side-effect of this is that explosions during the "manipulate organs" step will basically disembowel you, but it kinda fits anyway.

By the by, because of this, there is a tiny chance bluespace fishing rods can yield you random organs. Worry not, they're newly generated, so you won't be snatching it from another player by accident (at least for now).

## Why It's Good For The Game
It adds more possible weird and rare shenanigans involving surgery.

## Changelog

:cl:
Add: You can use a fishing rod to snatch organs during organ manipulation surgery
/:cl:

### CODE REVIEW

**Comment by SomeoneYouProbablyKnow** [comment]
Incredibly funny. Why skip the minigame?

File: code/modules/surgery/organ_manipulation.dm [hunk 1]
```diff
@@ -69,32 +101,6 @@
...
-	var/try_to_fail = FALSE
-	if(LAZYACCESS(modifiers, RIGHT_CLICK))
-		try_to_fail = TRUE
-
-	var/datum/surgery_step/step = get_surgery_step()
-	if(isnull(step))
-		return FALSE
-	var/obj/item/tool = user.get_active_held_item()
-	if(tool)
-		tool = tool.get_proxy_attacker_for(target, user)
```
Comment (SyncIt21) [inline code/modules/surgery/organ_manipulation.dm:89]: All these are important stuff for the borg omnitool to work. Did you test if the normal organ manipulation surgery works along with borgs?

**Comment by carlarctg** [comment]
you've gone too far this time

File: code/modules/surgery/organ_manipulation.dm (hunk 1 above)
Comment (Ghommie) [reply code/modules/surgery/organ_manipulation.dm:89]: Oh, at a glance, it looked just like standard, but I'll give it another look.

**Comment by Ghommie** [comment]
> Incredibly funny. Why skip the minigame?

Because it's limited to the "manipulate organs" step, which on average enough time to justify the minigame, otherwise it'd be fruitless an ovwerwhelmingly majority of the times.

File: code/modules/surgery/organ_manipulation.dm (hunk 1 above)
Comment (00-Steven) [reply code/modules/surgery/organ_manipulation.dm:89]: I believe this entire section above here and the two `try_op(...)` lines below are the same as the parent `next_step(...)`:
https://github.com/tgstation/tgstation/blob/7a7443090b8c23a44048fbbfe2fb3ca7cf908792/code/modules/surgery/surgery.dm#L111-L130
However, it looks like the tool check below is different from the parent:
https://github.com/tgstation/tgstation/blob/7a7443090b8c23a44048fbbfe2fb3ca7cf908792/code/modules/surgery/surgery.dm#L131-L135
Where it checks for whether the tool has a tool behaviour instead of being a surgical tool

I don't think removing this would cause issues with borg behaviour, as the parent still has the exact same proxy attacker code as shown above
But! As the comment below says, `//Mechanic organ manipulation isn't done with just surgery tools`, in comparison to the comment for the parallel check on the parent, `//Just because you used the wrong tool it doesn't mean you meant to whack the patient with it`, I assume this code was here to stop you from whacking people in the chest when using the wrong tools during mechanical organ manipulation surgery
So it may be good to look into that

Comment (Ghommie) [reply code/modules/surgery/organ_manipulation.dm:89]: Right, so the solution is to make a new proc (or var, flag w/e) that checks if we shouldn't wack the person with the tool we just used if it has the surgery tool flag for standard surgeries or if it has a tool behaviour for mechanical organ manipulation. Thanks for the help.

File: code/_globalvars/lists/objects.dm [hunk 2]
```diff
@@ -80,3 +84,16 @@ GLOBAL_LIST_EMPTY(roundstart_station_borgcharger_areas)
 
 /// List of area names of roundstart station mech rechargers, for the low charge/no charge mech screen alert tooltips.
 GLOBAL_LIST_EMPTY(roundstart_station_mechcharger_areas)
+
+// List of organ typepaths that are not unit test-able, and shouldn't be spawned by some things, such as certain class prototypes.
+GLOBAL_LIST_INIT(prototype_organs, typecacheof(list(
+	/obj/item/organ,
```
Comment (MrMelbert) [inline code/_globalvars/lists/objects.dm:90]: If I am not mistaken, you are making this typecache contain all organs in existence, because you are not calling it with `only_root_path = TRUE`. Is this intentional?

Comment (MrMelbert) [reply code/_globalvars/lists/objects.dm:90]: Oh I realize you just moved this from a static to a global, was this always broken? 😄

**Comment by san7890** [comment]
undraft when ci is passing

**Comment by norsvenska** [comment]
"John McGreytide, get out of the operating room. I'm replacing this guy's-- no, put the fishing rod away. Don't even thin-- GIVE ME THAT"

**Comment by Ghommie** [comment]
It turns out the `organs sanity` unit test was massively fucked, to the point I had been sidetracked into refactoring organ insertion and removal, clean a lot of hard deletes, fix several runtime issues with stomachs, fix other issues with bodypart overlays, blacklist some organs, curse at dullahan shitcode, and curse at BCIs too, for good measure. All of this simply because I dared to make a static list from the organ sanity test into a global list for organ-related fishing tomfoolery.

File: code/modules/antagonists/abductor/equipment/gland.dm [hunk 3]
```diff
@@ -93,10 +93,10 @@
 	hud.remove_atom_from_hud(gland_owner)
 	clear_mind_control()
 
-/obj/item/organ/heart/gland/mob_insert(mob/living/carbon/gland_owner, special = FALSE, movement_flags = DELETE_IF_REPLACED)
+/obj/item/organ/heart/gland/on_mob_insert(mob/living/carbon/gland_owner, special = FALSE, movement_flags)
 	. = ..()
 
-	if(special != 2 && uses) // Special 2 means abductor surgery
+	if(!special && uses) // Special 2 means abductor surgery
```
Comment (MrMelbert) [inline code/modules/antagonists/abductor/equipment/gland.dm:99]: You seem to have broken this code; the comment is true

image

You should replace it with a define

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/surgery/organs/internal/eyes/_eyes.dm [hunk 4]
```diff
@@ -550,7 +550,7 @@
 	deactivate(close_ui = TRUE)
 
 /// Set the initial color of the eyes on insert to be the mob's previous eye color.
-/obj/item/organ/eyes/robotic/glow/mob_insert(mob/living/carbon/eye_recipient, special = FALSE, movement_flags = DELETE_IF_REPLACED)
+/obj/item/organ/eyes/robotic/glow/on_mob_insert(mob/living/carbon/eye_recipient, special = FALSE, movement_flags)
```
Comment (MrMelbert) [inline code/modules/surgery/organs/internal/eyes/_eyes.dm:553]: In some places, you have defined `on_mob_insert` twice: 

553: /obj/item/organ/eyes/robotic/glow/on_mob_insert
559: /obj/item/organ/eyes/robotic/glow/on_mob_insert

File: code/modules/antagonists/abductor/equipment/gland.dm (hunk 3 above)
Comment (Ghommie) [reply code/modules/antagonists/abductor/equipment/gland.dm:99]: That should be an organ movement flag but sure.
